src/passes/analysis/verilog.cpp

```diff
@@ -1594,14 +1600,60 @@ bool Passes::Verilog::runOnInstanceGraphNode(InstanceGraphNode& node) {
   return false;
 }
 
+bool add_prefix(
```

Comment:
is there any reason to do this as a pass rather than at code generation time? I like this more as a code-generation option than a pass.

one of the reasons for this is that it makes symbol-tracking easier, if it's done "on site" rather than as a post-pass.

Reply:
It's complicated to do this at code generation time, since there are a lot of different places where names are computed (e.g. different logic for the primitives versus standard modules versus modules coming from verilog strings).  It's doable, but doing it after the standard names are created is much simpler.  Could we just update the symbol table at this step since we can just map the non-prefixed names to the new prefix names?

Reply:
Yeah on second thought, I think it's fine to do as a pass.